pilot/pkg/xds/xdsgen.go

```diff
@@ -134,12 +133,6 @@ func (s *DiscoveryServer) pushXds(con *Connection, w *model.WatchedResource, req
 			log.Debugf("%s: SKIP%s for node:%s%s", v3.GetShortType(w.TypeUrl), req.PushReason(), con.proxy.ID, info)
 		}
 
-		// If we are sending a request, we must respond or we can get Envoy stuck. Assert we do.
-		// One exception is if Envoy is simply unsubscribing from some resources, in which case we can skip.
-		isUnsubscribe := !req.Delta.IsEmpty() && req.Delta.Subscribed.IsEmpty()
-		if features.EnableUnsafeAssertions && err == nil && res == nil && req.IsRequest() && !isUnsubscribe {
```

Comment:
not needed because now we donot  respond for this kind of xds request

Reply:
Not sure I understand this. Isn't the assertion saying: "If we do not respond to this request, Envoy will get stuck"?

So if we did actually stop responding, then we broke it? Unless its no longer true

Reply:
Basically this is removing a test case/assertion... why do we not want this? We seem to be losing test coverage and could have a regression here

Reply:
See original PR https://github.com/istio/istio/commit/a52abd89c79801ad7b7961afc7a9e8b03fb40db8

Reply:
I understand the intention. Here i do make the check before in `shouldRespond` and `shouldRespondDelta`., rather than during the push XDS

FYI in shouldRespond `if len(removed) == 0 && len(added) == 0 {` we mark it as an ack